### PR TITLE
DPAT-2026 Establish project structure

### PR DESCRIPTION
**Pull request by ymao2**
- Add initial project structure
- Update python dependencies
- Add pre-commit config to run linting tools
- Update dev container to install and run postgres, install dependencies and initialise pre-commit

### CODE REVIEW

**Comment by Gary-H9** [comment]
Updated devcontainer to include a postgresql , the configuration is in [here](https://github.com/ministryofjustice/data-platform-control-panel/blob/dpat-2062-tech-structure/contrib/docker-compose-postgres.yml). Username and password are configured in there and can be changed if desired. Additionally a [feature](https://github.com/ministryofjustice/data-platform-control-panel/blob/e201cc64dc0093fe422e4b3af2f95c94fc975385/.devcontainer/devcontainer.json#L11) was added to the devcontainer which runs [Docker in Docker](https://github.com/microsoft/vscode-dev-containers/blob/main/containers/docker-in-docker/README.md). From a users perspective this means that `postgresql` is run in a container inside the devcontainer, and you can see the container running on the standard port `5432`. You can connect to the container using the command: 
```
psql -h localhost -U controlpanel
```

**Comment by michaeljcollinsuk** [comment]
Good starting point 👷‍♀️

**Review by michaeljcollinsuk — APPROVED** [review]

File: README.md
Comment (jacobwoffenden) [inline README.md:1]: Can you move `## Running` below the compliance badge?

**Review by jacobwoffenden — CHANGES_REQUESTED** [review]

File: README.md
Comment (michaeljcollinsuk) [reply README.md:1]: Good catch, done

**Review by michaeljcollinsuk — DISMISSED** [review]

**Review by michaeljcollinsuk — DISMISSED** [review]

**Review by jacobwoffenden — APPROVED** [review]

**Review by michaeljcollinsuk — APPROVED** [review]

**Pull request merged**